src/components/preview/preview.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+function Preview (props) {
+  const isLoading = (() => {
+    if (typeof props.loading !== 'undefined') {
+      return props.loading;
+    }
+    return !props.children;
+  })();
+
+  const mainClasses = (() => {
+    return classNames(
+      'carbon-preview',
+      { 'carbon-preview--placeholder': isLoading },
+      props.className
+    );
+  })();
+
+  return (
+    <div className={ mainClasses }>
+      { props.children }
+    </div>
+  );
+}
+
+Preview.propTypes = {
+  /**
+  * Children elements
+  *
+  * @property children
+  * @type {Node}
+  */
+  children: PropTypes.node,
```

Comment:
Should we make `children` a required prop? Otherwise without any `children` we've got a component that renders an empty `<div>`.

Reply:
It's not required. It's just possible prop.

Reply:
Oops, my mistake - I mis-interpreted where the `placeholder` CSS was applied. For some reason I thought it was applied to the `children` but it's on the `<div>`.